src/custom_linear_maps.jl

```diff
@@ -66,17 +66,9 @@ LinearAlgebra.adjoint(A::PDLinMapWithChol) = A
 LinearAlgebra.transpose(A::PDLinMapWithChol{<:Real}) = A
 LinearAlgebra.transpose(A::PDLinMapWithChol) = PDLinMapWithChol(transpose(A.parent), A.chol_L)
 
-Base.@propagate_inbounds LinearMaps.A_mul_B!(
+Base.@propagate_inbounds LinearMaps._unsafe_mul!(
```

Comment:
Unfortunately, this is dangerous: `_unsafe_mul!` is an internal function of LinearMaps, and not covered by semantic versioning. So it could change or vanish at any time, breaking MGVI.

However, I think we may not need the `A..._mul_B` functions (anymore) - can you try just taking them out, leaving only the `LinearAlgebra.mul!` definitions in?

Reply:
Well spotted, thank you!! Indeed it does work without unsafe overload.